week_2/role.tf

```diff
@@ -0,0 +1,44 @@
+resource "aws_iam_role_policy" "s3_full_access_policy" {
+  name = "s3_full_access_policy"
+  role = aws_iam_role.ec2_s3_full_access_role.id
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : [
+          "s3:*"
```

Comment:
nit: fiving full access is a bad practice, you should try to give minimal permissions needed for the instance, i.e. List and GetObject from s3.

Reply:
- changed s3 access to getObject only